### PR TITLE
Polars parquet  #417

### PR DESCRIPTION
**Pull request by swapdewalkar**
Polars Parquet Data Saver /Loader.
Implement https://pola-rs.github.io/polars/py-polars/html/reference/io.html#parquet as a data saver/loader.

## Changes
- Parquet Wrapper
- Unit tests
- Materialize Example

## How I tested this
I ran unit tests.

## Checklist

- [x] PR has an informative and human-readable title (this will be pulled into the release notes)
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future TODOs are captured in comments
- [x] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: examples/polars/materialization/my_script.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        id="df_to_feather",
+        path="./df.feather",
+        combine=df_builder,
+    ),    
+    to.parquet(
+        dependencies=output_columns,
+        id="df_to_parquet",
+        path="./df.parquet",
+        combine=df_builder,
+    ),
```
Comment (skrawcz) [inline examples/polars/materialization/my_script.py:99]: I think the only one that works is `to.parquet` (which you're adding) for polars.

File: examples/polars/materialization/my_script.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    *materializers,
+    additional_vars=[
+        "df_to_pickle_build_result",
+        "df_to_json_build_result",
+        "df_to_sql_build_result",
+        "df_to_xml_build_result",
+        "df_to_html_build_result",
+        "df_to_stata_build_result",
+        "df_to_feather_build_result",
+        "df_to_parquet_build_result",
```
Comment (skrawcz) [inline examples/polars/materialization/my_script.py:120]: adjust this to be just for polars.

File: examples/polars/materialization/my_script.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+)
+print(materialization_results)
+print(additional_outputs["df_to_pickle_build_result"])
+print(additional_outputs["df_to_json_build_result"])
+print(additional_outputs["df_to_sql_build_result"])
+print(additional_outputs["df_to_xml_build_result"])
+print(additional_outputs["df_to_html_build_result"])
+print(additional_outputs["df_to_stata_build_result"])
+print(additional_outputs["df_to_feather_build_result"])
+print(additional_outputs["df_to_parquet_build_result"])
```
Comment (skrawcz) [inline examples/polars/materialization/my_script.py:132]: and then this part too.

**Comment by skrawcz** [comment]
>     storage_options: dict[str, Any] = None
> E   TypeError: 'type' object is not subscriptable

For python 3.7 -- `dict[str, Any]` isn't valid. Instead it should be `Dict[str, Any]`.

File: examples/polars/materialization/notebook.ipynb [hunk 4]
```diff
@@ -0,0 +1,899 @@
...
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "materializers = [\n",
+    "    # materialize the dataframe to a parquet file\n",
+    "    to.parquet(\n",
+    "        dependencies=output_columns,\n",
+    "        id=\"df_to_pickle\",\n",
+    "        path=\"./df.pkl\",\n",
```
Comment (skrawcz) [inline examples/polars/materialization/notebook.ipynb:703]: these should be updated for parquet.

File: examples/polars/materialization/my_script.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+# set up db connection for sql materializer below
+conn = sqlite3.connect("df.db")
+
+# remove an previous instances of the 'test' table that will be created next
+conn.cursor().execute("DROP TABLE IF EXISTS test;")
+conn.commit()
+
+materializers = [
+    # materialize the dataframe to a pickle file
```
Comment (skrawcz) [inline examples/polars/materialization/my_script.py:50]: change comment to parquet

**Review by skrawcz — COMMENTED** [review]
almost there

**Comment by swapdewalkar** [comment]
@skrawcz update take a look once!

File: examples/polars/materialization/my_script.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
+import logging
+import sqlite3
```
Comment (skrawcz) [inline examples/polars/materialization/my_script.py:2]:
```suggestion
```
remove.

File: examples/polars/materialization/my_script.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    "spend_per_signup",
+    "spend_zero_mean_unit_variance",
+]
+
+# set up db connection for sql materializer below
+conn = sqlite3.connect("df.db")
+
+# remove an previous instances of the 'test' table that will be created next
+conn.cursor().execute("DROP TABLE IF EXISTS test;")
+conn.commit()
```
Comment (skrawcz) [inline examples/polars/materialization/my_script.py:47]:
```suggestion
```
remove

File: examples/polars/materialization/my_script.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    *materializers,
+    additional_vars=[
+        "df_to_parquet_build_result",
+    ],  # because combine is used, we can get that result here.
+    inputs=initial_columns,
+)
+print(materialization_results)
+print(additional_outputs["df_to_parquet_build_result"])
+
+conn.close()
```
Comment (skrawcz) [inline examples/polars/materialization/my_script.py:77]:
```suggestion
```
remove

File: examples/polars/materialization/notebook.ipynb [hunk 9]
```diff
@@ -0,0 +1,899 @@
...
+   "execution_count": 2,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2023-09-17T05:43:33.890006Z",
+     "start_time": "2023-09-17T05:43:33.866292Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import sqlite3\n",
```
Comment (skrawcz) [inline examples/polars/materialization/notebook.ipynb:38]:
```suggestion
```

File: examples/polars/materialization/notebook.ipynb [hunk 10]
```diff
@@ -0,0 +1,899 @@
...
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# set up db connection for sql materializer below\n",
+    "conn = sqlite3.connect(\"df.db\")\n",
+    "\n",
+    "# remove an previous instances of the 'test' table that will be created next\n",
+    "conn.cursor().execute(\"DROP TABLE IF EXISTS test;\")\n",
+    "conn.commit()"
```
Comment (skrawcz) [inline examples/polars/materialization/notebook.ipynb:220]:
```suggestion
    "# pass"
```

File: examples/polars/materialization/notebook.ipynb [hunk 11]
```diff
@@ -0,0 +1,899 @@
...
+    "additional_outputs[\"df_to_parquet_build_result\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "conn.close()"
```
Comment (skrawcz) [inline examples/polars/materialization/notebook.ipynb:874]:
```suggestion
    "#pass"
```

File: examples/polars/materialization/requirements.txt [hunk 12]
```diff
@@ -0,0 +1,3 @@
+lxml # required for XML
```
Comment (skrawcz) [inline examples/polars/materialization/requirements.txt:1]:
```suggestion
```
remove

**Review by skrawcz — APPROVED** [review]
Looks good -- just need to remove references to things we don't need yet.

**Pull request merged**